receiver/k8sclusterreceiver/internal/node/nodes.go

```diff
@@ -52,79 +48,56 @@ func Transform(node *corev1.Node) *corev1.Node {
 	return newNode
 }
 
-func GetMetrics(node *corev1.Node, nodeConditionTypesToReport, allocatableTypesToReport []string, logger *zap.Logger) []*agentmetricspb.ExportMetricsServiceRequest {
-	metrics := make([]*metricspb.Metric, 0, len(nodeConditionTypesToReport)+len(allocatableTypesToReport))
+func GetMetrics(set receiver.CreateSettings, node *corev1.Node, nodeConditionTypesToReport, allocatableTypesToReport []string) pmetric.Metrics {
+	mb := imetadata.NewMetricsBuilder(imetadata.DefaultMetricsBuilderConfig(), set)
+	ts := pcommon.NewTimestampFromTime(time.Now())
+
 	// Adding 'node condition type' metrics
 	for _, nodeConditionTypeValue := range nodeConditionTypesToReport {
-		nodeConditionMetric := getNodeConditionMetric(nodeConditionTypeValue)
-		v1NodeConditionTypeValue := corev1.NodeConditionType(nodeConditionTypeValue)
-
-		metrics = append(metrics, &metricspb.Metric{
-			MetricDescriptor: &metricspb.MetricDescriptor{
-				Name: nodeConditionMetric,
-				Description: fmt.Sprintf("Whether this node is %s (1), "+
-					"not %s (0) or in an unknown state (-1)", nodeConditionTypeValue, nodeConditionTypeValue),
-				Type: metricspb.MetricDescriptor_GAUGE_INT64,
-			},
-			Timeseries: []*metricspb.TimeSeries{
-				utils.GetInt64TimeSeries(nodeConditionValue(node, v1NodeConditionTypeValue)),
-			},
-		})
+		switch nodeConditionTypeValue {
+		case "Ready":
+			v1NodeConditionTypeValue := corev1.NodeConditionType(nodeConditionTypeValue)
```

Comment:
Can this be moved outside of the switch block to avoid repeating?

Reply:
done, and using that for the switch to avoid magic strings.